cli/core/pkg/command/config.go

```diff
@@ -216,39 +216,29 @@ var initConfigCmd = &cobra.Command{
 		if cfg.ClientOptions.CLI == nil {
 			cfg.ClientOptions.CLI = &configapi.CLIOptions{}
 		}
-		repos := cfg.ClientOptions.CLI.Repositories
-		finalRepos := []configapi.PluginRepository{}
-		for _, repo := range config.DefaultRepositories {
-			var exists bool
-			for _, r := range repos {
-				if repo.GCPPluginRepository.Name == r.GCPPluginRepository.Name {
-					finalRepos = append(finalRepos, r)
-					exists = true
-				}
-			}
-			if !exists {
-				finalRepos = append(finalRepos, repo)
-			}
+
+		serverName := ""
+		server, err := cfg.GetCurrentServer()
+		if err == nil && server != nil {
+			serverName = server.Name
 		}
-		cfg.ClientOptions.CLI.Repositories = finalRepos
 
-		err = configlib.StoreClientConfig(cfg)
+		serverPluginDescriptors, standalonePluginDescriptors, err := pluginmanager.InstalledPlugins(serverName)
 		if err != nil {
 			return err
 		}
-		// TODO: cli.ListPlugins is deprecated: Use pluginmanager.AvailablePluginsFromLocalSource or pluginmanager.AvailablePlugins instead
-		descriptors, err := cli.ListPlugins()
+
+		for _, desc := range append(serverPluginDescriptors, standalonePluginDescriptors...) {
+			config.AddDefaultFeatureFlagsIfMissing(cfg, desc.DefaultFeatureFlags)
+		}
+
+		err = configlib.StoreClientConfig(cfg)
 		if err != nil {
 			return err
 		}
 
-		errList := []error{}
-		for _, desc := range descriptors {
-			if err := cli.InitializePlugin(desc.Name); err != nil {
```

Comment:
This seems to be a change in behaviour where `tanzu config init` will no longer initialize all the plugins.
I guess that is ok?

Reply:
It looks like a change in behavior but it is actually not considering the bug in implementation here where we always return empty `descriptors` array because of missing implementation of `context-aware-cli-for-plugins` featureflag enabled.

Reply:
Aha 😁.  Then nice cleanup 👍